proposals/v3/api.md

```diff
@@ -0,0 +1,815 @@
+There may be 0..N health checks for a Plugin, depending on how it is configured.
+The health check elements here make up a snapshot of the plugin's health at a given time.
+
+| Field | Description |
+| ----- | ----------- |
+| *name* | The name of the health check. |
+| *status* | The status of the health check (ok, failing) |
+| *message* | A message describing the failure, if in a failing state. |
+| *timestamp* | The timestamp for which the status applies. |
+| *type* | The type of health check (e.g. periodic) |
+
+#### Error (500)
+See: [Errors](#errors)
+
+
+
+## Scan
+```
+GET http://HOST:5000/synse/v3/scan
+```
+
+List the devices that Synse knows about and can read from/write to via
+the configured plugins.
+
+This endpoint provides an aggregated view of the devices made known to Synse
+Server by each of the registered plugins. This endpoint provides a high-level
+view of what exists in the system. Scan info can be filtered to show only those
+devices which match a set of provided tags.
+
+The scan results are sorted by device id.
+
+### Query Parameters
+
+| Key | Description |
+| --- | ----------- |
+| namespace | The default namespace to use for the specified labels. (default: `default`) |
+| ns | Same as `namespace`, above. |
+| tags | The [tags](tags.md) to filter devices on. If specifying multiple tags, they should be comma-separated. |
+| force | Force a re-scan (do not use the cache). This will take longer than scanning using the cache, since it needs to rebuild the cache. (default: false) |
+| sort | Specify the fields to sort by. Multiple fields can be specified as a comma separated string, e.g. `"plugin,id"`. (default: "id") |
+
+### Response
+
+#### OK (200)
+```json
+[
+  {
+    "id": "0fe8f06229aa9a01ef6032d1ddaf18a5",
+    "info": "Synse Temperature Sensor",
+    "type": "temperature",
+    "plugin": "12835beffd3e6c603aa4dd92127707b5",
+    "tags": [
+      "type:temperature",
+      "temperature",
+      "vio/fan-sensor"
+    ]
+  },
+  {
+    "id": "12ea5644d052c6bf1bca3c9864fd8a44",
+    "info": "Synse LED",
+    "type": "led",
+    "plugin": "12835beffd3e6c603aa4dd92127707b5",
+    "tags": [
+      "type:led",
+      "led"
+    ]
+  }
+]
+```
+
+#### Error (500, 400)
+See: [Errors](#errors)
+
+
+
+## Health
+```
+GET http://HOST:5000/synse/v3/health
+```
+
+*See also: [Health](health.md)*
+
+### Response
+
+#### OK (200)
+
+#### Error (500)
+See: [Errors](#errors)
+
+
+## Capabilities?
+> Not sure this endpoint is needed, but can't hurt to have...
+
+```
+GET http://HOST:5000/synse/v3/capabilities
+```
+
+### Response
+
+#### OK (200)
+
+#### Error (500)
+See: [Errors](#errors)
+
+
+
+## Info
+> **Changes**: There is no longer a need for a distinction between rack/board/device
+> info -- its all just device info now.
+
+```
+GET http://HOST:5000/synse/v3/info/{device_id}
+```
+
+Get the full set of metainfo for a specified device.
+
+> **Question**: Should we include additional information here, e.g. whether the device
```

Comment:
the device capabilities is defined above this section, but isn't filled in yet (since I'm not sure its totally needed). It does exist in the current version as well: https://vapor-ware.github.io/synse-server/#capabilities

I do think we need that additional info, but I'm just not sure whether to compartmentalize things (e.g. metainfo from the /info endpoint, and capabilities from the /capabilities endpoint) or to just have it all together. The info endpoint kinda already shows some of the capabilities by listing the outputs the device supports

Reply:
Ah I see. Since the current `/info` shows the capabilities already in details already, the `/capabilities` feels somewhat unnecessary. In that case, I agree that the `capabilities` is not really needed anymore.

For additional information, like device support, I think it's fine to add in one more field in the info scheme and have all truth for a single device there.

Reply:
👍 

The info encoded in capabilities has moved to /info